src/components/NHeading/NHeading.vue

```diff
@@ -0,0 +1,89 @@
+<template>
+  <component :is="tag" :style="customStyle" :class="sizeClass" v-bind="$attrs" v-on="$listeners">
+    <slot></slot>
+  </component>
+</template>
+
+  
+<script>
+export default {
+  name: "Heading",
+  props: {
+    level: {
+      type: String,
+      default: "h2",
+      validator: (value) => ["h1", "h2", "h3", "h4", "h5", "h6"].includes(value),
+    },
+    color: {
+      type: String,
+      default: "#222",
+    },
+    fontSize: {
+      type: String,
+      default: "",
+    },
+    size: {
+      type: String,
+      default: "", 
+    },
+  },
+  computed: {
+    tag() {
+      return this.level;
+    },
+    customStyle() {
+      return {
+        color: this.color,
+        fontSize: this.fontSize,
+      };
+    },
+    sizeClass() {
+      // Compute the size class based on the "size" prop
+      return this.size ? `size-${this.size.toLowerCase()}` : '';
+    },
+  },
+};
+</script>
+
+
+
+<style lang="less">
+
+.size-l{
+  font-size: 40px;
+  letter-spacing: -1.44px !important;
```

Comment:
it's not good practice to use !important in component library

Reply:
i will update